decision-records/010-remove-init-method.md

```diff
@@ -0,0 +1,98 @@
+# Remove `init()` method of JavaScript components
+<!-- The title should be a short present tense imperative phrase, less than 50
+     characters, like a git commit message. -->
+
+**Date:** 2023-07-20
+
+**Status:** Accepted
+
+## Decision
+
+<!-- An overview of the decision made. -->
+
+We'll be removing the `init()` method of our JavaScript components,
+in favour of them initialising automatically during construction.
+
+This simplifies individual component's initialisation from:
+
+```js
+new Accordion($element).init()
+```
+
+To:
+
+```js
+new Accordion($element)
+```
+
+## Rationale
+
+The `init()` method adds an extra step to our components initialisation,
+without tangible benefits for what our components do. Without it we can simplify both how our components are initialised, as well as their internal implementation as we're looking to [throw errors at construction](./009-throw-errors-component-constructors.md).
+
+The `init()` method enables the following, that none of our components benefit from, so we can remove that extra step:
+
+- delaying the initialisation of the component after its instantiation
+- allowing the initialisation code to be `async` (which a `constructor` cannot be)
+
+With our intent to [throw errors at construction](./009-throw-errors-component-constructors.md), having a separate `init()` method requires careful separation between validation of the component in the `constructor()` and initialisation code in a subsequent call. By having only the constructor, the initialisation code can simply `throw` from within `constructor()` to ensure users always get functional instances of our components. 
+
+<!-- The reason for the decision and any comments or disagreements cited. -->
+
+## Risks and constraints
+
+### Reduce or delay the adoption of v5
+
+This change requires code changes from our users, that they may not have the capacity to implement.
+
+This is mitigated by the changes being quite small (removing a `.init()` call following the current instantiation of the component) and only affects a portion of our users, who use our component classes directly rather than through `initAll`.
```

Comment:
We talk about the impact on `init()` not `initAll()` further down too?

```suggestion
This is mitigated by the changes being quite small (removing a `.init()` call following the current instantiation of the component) and only affects a portion of our users, who use our component classes directly rather than through `initAll()`.
```

Reply:
Indeed, I'll add a link to the section down below 😊